### PR TITLE
Bump pip minimum version to >= 20.3

### PR DESCRIPTION
**Pull request by atugushev**
Due to `pip-tools`' [policy](https://github.com/jazzband/pip-tools/issues/926) to support only the last two minor pip versions, it's time to drop the `pip-20.1` and `pip-20.2`.

Latest: `21.0.*`
Previous: `20.3.*`

**Changelog-friendly one-liner**: Bump `pip` minimum version to `>= 20.3`.

##### Contributor checklist

- [ ] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: tests/test_resolver.py [hunk 1]
```diff
@@ -329,35 +328,23 @@ def test_compile_failure_shows_provenance(resolver, from_line):
...
             "test_package>=1.2",
             "test_package>=1.2.0",
             True,
-            marks=pytest.mark.skipif(
-                PIP_VERSION[:2] < (20, 2), reason="Required only for pip>=20.2"
-            ),
         ),
         ("test_package[foo,bar]==1.2", "test_package[bar,foo]==1.2", True),
         ("test_package[foo,bar]>=1.2", "test_package[bar,foo]>=1.2", True),
         pytest.param(
```
Comment (jdufresne) [inline tests/test_resolver.py:339]: Now that `marks` isn't required, can drop `pytest.param` and use a tuple (like the other lines). This pattern appears multiple times and could be cleaned up throughout the diff.

**Review by jdufresne — APPROVED** [review]
Only comment is very minor and cosmetic. Otherwise, looks great. Thanks for cleaning this up!

File: tests/test_resolver.py (hunk 1 above)
Comment (atugushev) [reply tests/test_resolver.py]: Thanks for noticing! Addressed in d9ab55df38221ef8c0f0af237d095c0ce3e65db1.

Comment (jdufresne) [reply tests/test_resolver.py]: :+1: LGTM

**Comment by atugushev** [comment]
Thanks @jdufresne for reviewing this! Much appreciated 👍🏼

**Pull request merged**